src/Yoti.Auth/ShareUrl/Policy/DynamicPolicyBuilder.cs

```diff
@@ -13,100 +13,108 @@ public DynamicPolicyBuilder WithWantedAttribute(WantedAttribute wantedAttribute)
             Validation.NotNull(wantedAttribute, nameof(wantedAttribute));
 
             string key = wantedAttribute.Derivation ?? wantedAttribute.Name;
+
+            if (wantedAttribute.Constraints?.Count > 0)
+            {
+                key += "-" + wantedAttribute.Constraints.GetHashCode();
```

Comment:
It would probably have made more sense to go with the multiple attribute changes, but I think I'll leave it here for now as it's still slightly related to these changes.